Assets/Plugins/iOS/Shopify/Deserializable/Deserializable.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  Deserializable.swift
+//  UnityBuySDK
+//
+//  Created by Shopify.
+//  Copyright © 2017 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+protocol Deserializable {
+    static func deserialized(_ serialized: JSON) -> Self?
+}
+
+extension Deserializable {
+    
+    static func deserialized(_ serialized: String) -> Self? {
+        guard
+            let data = serialized.data(using: .utf8),
+            let jsonObject = (try? JSONSerialization.jsonObject(with: data)) as? JSON
+        else {
+            return nil
+        }
+
+        return deserialized(jsonObject)
+    }
+    
+    static func items(fromSerialized jsons: [JSON]) -> [Self]? {
```

Comment:
Why not continue following the naming convention here? After all, a collection of `JSON` objects is still a "serialized value":
```swift
static func deserialize(_ jsonCollection: [JSON]) -> [Self]?
```